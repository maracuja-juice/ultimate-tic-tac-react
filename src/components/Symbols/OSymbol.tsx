import * as React from 'react';
import './symbols.css';
import './symbolAnimations.css';
import '../../fonts/icons.css';
import * as classNames from 'classnames';

interface OSymbolProps {
  bigSymbol?: boolean;
  shouldAnimate?: boolean;
  className?: string;
}

export class OSymbol extends React.Component<OSymbolProps> {
  public static defaultProps: Partial<OSymbolProps> = {
    shouldAnimate: true,
  };

  render() {
    const { bigSymbol, shouldAnimate, className } = this.props;
    const iconClass = classNames({
      'icon-o': true,
      o: true,
      'big-symbol': bigSymbol,
      'animate-o': shouldAnimate,
      className: className,
    });

    return <i className={iconClass} aria-label="Circle Symbol" />;
  }
}
