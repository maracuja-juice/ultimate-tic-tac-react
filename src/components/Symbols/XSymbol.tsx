import * as React from 'react';
import classNames from 'classnames';
import styles from './Symbols.module.css';
import { XGameIcon } from '../Icons';

export interface XSymbolProps {
  shouldAnimate?: boolean;
}

export class XSymbol extends React.Component<XSymbolProps> {
  public static defaultProps: Partial<XSymbolProps> = {
    shouldAnimate: true,
  };

  render() {
    const { shouldAnimate } = this.props;
    const iconClass = classNames({
      [styles.x]: true,
      [styles.animateX]: shouldAnimate,
    });

    return <XGameIcon className={iconClass} />;
  }
}
