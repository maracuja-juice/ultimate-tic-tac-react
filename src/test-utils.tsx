// See: https://testing-library.com/docs/react-testing-library/setup#custom-render
import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { IconProvider } from './components/Icons';
import '@testing-library/jest-dom'; // https://github.com/testing-library/jest-dom/issues/282#issuecomment-660034778
import { MemoryRouter } from 'react-router-dom';

const AllTheProviders: FC = ({ children }) => {
  return (
    <IconProvider>
      <MemoryRouter>{children}</MemoryRouter>
    </IconProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
