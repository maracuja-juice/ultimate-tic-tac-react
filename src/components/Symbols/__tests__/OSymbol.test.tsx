import * as React from 'react';
import { render } from '../../../test-utils';
import { OSymbol } from '../OSymbol';
import { XSymbol } from '../XSymbol';

test('renders o', () => {
  // This is a hack and should not be used in tests where we can queryByRole or other!
  const { container } = render(<OSymbol />);
  const o = container.querySelector('svg');
  expect(o).not.toBeNull();
  expect(o!.classList.value).toMatch(/animateo/i);
});

test('renders o animated', () => {
  // This is a hack and should not be used in tests where we can queryByRole or other!
  const { container } = render(<OSymbol shouldAnimate={true} />);
  const o = container.querySelector('svg');
  expect(o).not.toBeNull();
  expect(o!.classList.value).toMatch(/animateo/i);
});

test('renders o not animated', () => {
  // This is a hack and should not be used in tests where we can queryByRole or other!
  const { container } = render(<XSymbol shouldAnimate={false} />);
  const o = container.querySelector('svg');
  expect(o).not.toBeNull();
  expect(o!.classList.value).not.toMatch(/animateo/i);
});
