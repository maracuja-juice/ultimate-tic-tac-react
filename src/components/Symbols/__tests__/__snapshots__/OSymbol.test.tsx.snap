// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OSymbol should match snapshot 1`] = `
<ForwardRef(SvgO)
  aria-label="O"
  className="o animateO"
/>
`;

exports[`OSymbol should match snapshot 2`] = `
<ForwardRef(SvgO)
  aria-label="O"
  className="o"
/>
`;
